todoman/cli.py

```diff
@@ -115,33 +149,23 @@ def command_wrap(*a, **kw):
               'remain stable regardless of configuration or version.')
 @click.pass_context
 @click.version_option(prog_name='todoman')
-def cli(ctx, color, porcelain):
+def cli(click_ctx, color, porcelain):
+    ctx = click_ctx.ensure_object(AppContext)
     try:
-        config = load_config()
+        ctx.config = load_config()
     except ConfigurationException as e:
         raise click.ClickException(e.args[0])
 
-    ctx.obj = {
-        'config': config,
-    }
-
-    if porcelain:
-        ctx.obj['formatter'] = PorcelainFormatter()
-    else:
-        ctx.obj['formatter'] = TodoFormatter(
-            config['main']['date_format'],
-            config['main']['time_format'],
-            config['main']['dt_separator'],
-        )
-
-    color = color or config['main']['color']
+    color = color or ctx.config['main']['color']
     if color == 'always':
-        ctx.color = True
+        click_ctx.color = True
     elif color == 'never':
-        ctx.color = False
+        click_ctx.color = False
+
+    ctx.porcelain = porcelain
```

Comment:
There's a common interface both classes implement, doesn't mean that all methods have to have the same signature.

You gain something from having the same instancemethods on both classes because that's what TodoEditor etc call, but you don't gain anything from doing this at init time. The code doesn't get significantly shorter.